src/services/homework/hooks/submissions.js

```diff
@@ -376,10 +376,43 @@ const hasEditPermission = (context) => {
 	return Promise.reject(new Forbidden());
 };
 
-const hasDeletePermission = (context) => {
-	if (context.data.isTeamMember) {
+const isAssigned = async (context, homeworkId, userId) => {
+	const homeworkService = context.app.service('/homework');
+	const homework = await homeworkService.get(homeworkId);
```

Comment:
Make it sense to avoid passing context and the await request? 
If you pass the homework instead of context, it can be syncon and is also a normal is check.